addons/website_sale_delivery/controllers/main.py

```diff
@@ -193,17 +193,28 @@ def express_checkout_process_shipping_address(self, partial_shipping_address):
                     name=_('Anonymous express checkout partner for order %s', order_sudo.name),
                 )
 
-        # Returns the list of develivery carrier available for the sale order.
+        # Returns the list of delivery carrier available for the sale order.
         return sorted([{
-            'id': carrier.id,
-            'name': carrier.name,
-            'description': carrier.website_description,
-            'minorAmount': payment_utils.to_minor_currency_units(
-                WebsiteSaleDelivery._get_rate(carrier, order_sudo, is_express_checkout_flow=True)['price'],
-                order_sudo.currency_id,
-            ),
-        } for carrier in order_sudo._get_delivery_methods()],
-        key=lambda carrier: carrier['minorAmount'])
+                'id': carrier.id,
+                'name': carrier.name,
+                'description': carrier.website_description,
+                'minorAmount': payment_utils.to_minor_currency_units(price, order_sudo.currency_id),
+            } for carrier, price in WebsiteSaleDelivery._get_carriers_express_checkout(order_sudo).items()
+            ], key=lambda carrier: carrier['minorAmount'])
+
+    @staticmethod
+    def _get_carriers_express_checkout(order):
+        """ Return available carriers and their prices for the given order in dict of
+        {carrier: price}. """
```

Comment:
nitpicking: Could you elaborate a bit the docstring? :angel: 

```suggestion
       """ Return available carriers and their prices for the given order.
       
        :param recordset order: [...], as a `sale.order` record.
        :rtype: dict
        :return: A dict with a `delivery.carrier` recordset a key, and a rate shipment price as value.
        """
```